server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/QueryExecution.java

```diff
@@ -280,18 +281,30 @@ private void releaseResource() {
    * implemented with DataStreamManager)
    */
   @Override
-  public Optional<TsBlock> getBatchResult() {
+  public Optional<TsBlock> getBatchResult() throws IoTDBException {
+    checkArgument(resultHandle != null, "ResultHandle in Coordinator should be init firstly.");
     // iterate until we get a non-nullable TsBlock or result is finished
     while (true) {
       try {
-        if (resultHandle == null || resultHandle.isAborted() || resultHandle.isFinished()) {
+        if (resultHandle.isAborted()) {
+          logger.info("resultHandle for client is aborted");
+          stateMachine.transitionToAborted();
```

Comment:
Does that the resultHandle is `aborted` mean the stateMachine is `Failed` ? 

I mean, the transitionToAborted won't take effect ?

Reply:
Yeap, I also think so.